lab_1/main.py

```diff
@@ -51,7 +126,82 @@ def create_language_profile(language: str, text: str, stop_words: list) -> dict
     :param stop_words: a list of stop words
     :return: a dictionary with three keys – name, freq, n_words
     """
-    pass
+
+    if not isinstance(language, str) or not isinstance(text, str) or not isinstance(stop_words, list):
+        return None
+
+    for i in top_words_en:
+        for en_word in freq_dict_en:
+            if i == en_word:
+                val_list = list(reversed(sorted(freq_dict_en.values())))
+                en_top_n = dict(zip(top_words_en, val_list))
+    for i in top_words_de:
+        for de_word in freq_dict_de:
+            if i == de_word:
+                val_list = list(reversed(sorted(freq_dict_de.values())))
+                de_top_n = dict(zip(top_words_de, val_list))
+    for i in top_words_unknown:
+        for unknown_word in freq_dict_unknown:
+            if i == unknown_word:
+                val_list = list(reversed(sorted(freq_dict_unknown.values())))
+                unknown_top_n = dict(zip(top_words_unknown, val_list))
+
+    en = {"name": language_en,
+          "freq": en_top_n,
+          "n_words": len(en_top_n)}
+
+    de = {"name": language_de,
+          "freq": de_top_n,
+          "n_words": len(de_top_n)}
+
+    unknown = {"name": language_unknown,
+               "freq": unknown_top_n,
+               "n_words": len(unknown_top_n)}
+
+    language_profile = {}
+
+    for word in stop_words:
+        if word in text:
+            language_profile = {"name": language}
+
+        language_val = list(language_profile.values())
+        en_val = list(en.values())
+        de_val = list(de.values())
+
+        for value in language_val:
+            if value in en_val:
+                en_profile = en
+                return en_profile
+            if value in de_val:
+                de_profile = de
+                return de_profile
+            else:
+                unknown_profile = unknown
+                return unknown_profile
+
+language_en = 'en'
+language_de = 'de'
+language_unknown = 'unknown'
+
+en_text = str(tokenize(en_text))
+de_text = str(tokenize(de_text))
+unknown_text = str(tokenize(unknown_text))
+
+en_stop_words = remove_stop_words(tokens_en, stop_words_en)
+de_stop_words = remove_stop_words(tokens_de, stop_words_de)
+unknown_stop_words = remove_stop_words(tokens_unknown, stop_words_unknown)
+
+top_words_en = get_top_n_words(freq_dict_en, top_n_en)
+top_words_de = get_top_n_words(freq_dict_de, top_n_de)
+top_words_unknown = get_top_n_words(freq_dict_unknown, top_n_unknown)
+
+top_n_en = len(get_top_n_words(freq_dict_en, top_n_en))
+top_n_de = len(get_top_n_words(freq_dict_de, top_n_de))
+top_n_unknown = len(get_top_n_words(freq_dict_unknown, top_n_unknown))
+
+freq_dict_en = calculate_frequencies(tokens_en)
+freq_dict_de = calculate_frequencies(tokens_de)
+freq_dict_unknown = calculate_frequencies(tokens_unknown)
```

Comment:
The `main.py` file should not contain function calls - you should check your implementation using the `start.py` file.